### ISSUE TITLE
Importing an exercise takes up large amounts of memory

### ISSUE THREAD

**Issue by ClFeSc** [hpi-sam/digital-fuesim-manv#542]
When importing an exercise to the database a huge amount of memory is taken up (at least on my laptop) until Node goes out of memory and crashes. It should be investigated
1. whether this occurs repeatedly on different machines, and
2. whether this is due to an incorrect/inefficient implementation or whether it's a direct result of the large exercise.

**Comment by anonym-HPI** [comment]
With database disabled also a timeout (local development, so can't be nginx timeouts) can happen and showing the big red warning message, but the upload after a while finishes and it is possible to enter the exercise.

**Comment by Greenscreen23** [comment]
What dataset did you use to test this? The ones from [digital-fuesim-manv_test-scenarios](https://github.com/hpi-sam/digital-fuesim-manv_test-scenarios) or a different exercise?

**Comment by ClFeSc** [comment]
I think it was the Ahrweiler scenario which should be included in the mentioned repo.

**Comment by anonym-HPI** [comment]
> I think it was the Ahrweiler scenario which should be included in the mentioned repo.

Yes should be Ahrweiler (BABZ großes Szenario), don't know if the one I tested is identical with the one in the test repo, but was (as far as I know) from the same exercise.

**Comment by Greenscreen23** [comment]
Btw, I wanted to verify this on my setup, and the docker containers index page doesn't look like the current page. It is missing the import button f.e. :D
I still have some issues building the container because `npm run deployment` is timing out in the hpi wifi, even after increasing the timeout. I'll try to build the container at home and verify this later.

**Comment by anonym-HPI** [comment]
> Btw, I wanted to verify this on my setup, and the docker containers index page doesn't look like the current page. It is missing the import button f.e. :D
> I still have some issues building the container because `npm run deployment` is timing out in the hpi wifi, even after increasing the timeout. I'll try to build the container at home and verify this later.

(Mind what I described happened locally without using docker and with database disabled - but what I described is another problem, as without database).

Missing the import button, do you mean the public image on hub.docker.com?

**Comment by Greenscreen23** [comment]
I wanted to test it with and without the database, as both cases were mentioned above. The image I used was `digitalfuesimmanv/dfm`

**Comment by anonym-HPI** [comment]
> I wanted to test it with and without the database, as both cases were mentioned above. The image I used was `digitalfuesimmanv/dfm:latest`

Did you test it locally without database and without docker?

**Comment by Greenscreen23** [comment]
Jup, that works just fine. No signs of node crashing or consuming too much memory

**Comment by Greenscreen23** [comment]
I've just managed to build the docker container, and I can't reproduce the issue mentioned. I can import and join all three scenarios both with and without a database using docker.

Are you guys still able to reproduce the issue on the latest commit of dev? Maybe this has been fixed somehow.

**Comment by ClFeSc** [comment]
I can retest it on my setups. Btw. you know that the `latest` tag on hub.docker.com is about half a year old, as no one has pushed to `main` since then? So it's expected to be outdated.

**Comment by Greenscreen23** [comment]
We could do a meeting for this if want. I'll message you in a private chat for this :)

**Comment by lukasrad02** [comment]
Could not reproduce with latest version
